src/main/java/homework_4/custom_file_reader/Main.java

```diff
@@ -0,0 +1,19 @@
+package homework_4.custom_file_reader;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+public class Main {
+    public static void main(String[] args) {
+        try {
```

Comment:
1. Better have one try-with-resources per one stream. So it's better to have 4 try/catch in run() methods.
2. You don't close streams

Reply:
Done.